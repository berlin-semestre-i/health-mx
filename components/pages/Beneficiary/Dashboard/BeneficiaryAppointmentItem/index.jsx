import React from 'react'
import { Grid, Image } from 'semantic-ui-react'
import styled from 'styled-components'
import media from 'styled-media-query'

const BeneficiaryAppointmentItem = ({ appointment }) => (
  <ConsultationRow>
    <AvatarContainer mobile={16} tablet={3} computer={2}>
      <Avatar src="../../../static/images/AppointmentIcon.png"/>
    </AvatarContainer>
    <ConsultationInfoContainer mobile={16} tablet={13} computer={14}>
      <Bold>{ appointment.name }</Bold>
      <ConsultationInfo>
        <label>{appointment.date}, {appointment.time} - {appointment.clinic}</label>
      </ConsultationInfo>
    </ConsultationInfoContainer>
  </ConsultationRow>
)

export default BeneficiaryAppointmentItem

const ConsultationRow = styled(Grid)`
  >.column:not(.row):last-child {
    padding-top: 1.5rem;
    margin-bottom: 1.5rem;
  }
`
const ConsultationInfo = styled.div`
  font-size: 14px;

  ${media.greaterThan('large')`
    width: 85%;
    display: flex;
    justify-content: space-between;
  `} 
`
const AvatarContainer = styled(Grid.Column)`
  text-align: center;

  ${media.between('medium','1025px')`
    [class*="two wide computer"]&&&.column {
      width: 18.75%!important;
    }
  `}

  ${media.lessThan('medium')`
    &&&.column:not(.row) {
      padding-bottom: 0;
    }
  `}
`
const ConsultationInfoContainer = styled(Grid.Column)`
  ${media.between('medium','1025px')`
    [class*="fourteen wide computer"]&&&.column {
      width: 81.25%!important;
    }
  `}
`
const Avatar = styled(Image)`
  && {
    border-radius: 30px;
    max-width: 55px;
    max-height: 55px;
  }
`
const Bold = styled.b`
  font-weight: 600;
`