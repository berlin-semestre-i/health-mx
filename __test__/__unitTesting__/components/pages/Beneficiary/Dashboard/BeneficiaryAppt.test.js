import React from 'react'
import { shallow, mount } from 'enzyme'
// eslint-disable-next-line max-len
import BeneficiaryAppointmentItem from '../../../../../../components/pages/Beneficiary/Dashboard/BeneficiaryAppointmentItem/index'

const testAppt = {
  name: 'Yulia',
  date: '10 Oct, 2018',
  time: '10:32am',
  clinic: 'General 2',
}

describe('Beneficiary Appointments', () => {
  it('renders', () => {
    const component = shallow(<BeneficiaryAppointmentItem appointment={testAppt} />)
    expect(component).toMatchSnapshot()
  })
})