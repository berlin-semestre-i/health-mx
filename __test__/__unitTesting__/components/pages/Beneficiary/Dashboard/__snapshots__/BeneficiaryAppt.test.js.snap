// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beneficiary Appointments renders 1`] = `
<BeneficiaryAppointmentItem__ConsultationRow>
  <BeneficiaryAppointmentItem__AvatarContainer
    computer={2}
    mobile={16}
    tablet={3}
  >
    <BeneficiaryAppointmentItem__Avatar
      src="../../../static/images/AppointmentIcon.png"
    />
  </BeneficiaryAppointmentItem__AvatarContainer>
  <BeneficiaryAppointmentItem__ConsultationInfoContainer
    computer={14}
    mobile={16}
    tablet={13}
  >
    <BeneficiaryAppointmentItem__Bold>
      Yulia
    </BeneficiaryAppointmentItem__Bold>
    <BeneficiaryAppointmentItem__ConsultationInfo>
      <label>
        10 Oct, 2018
        , 
        10:32am
         - 
        General 2
      </label>
    </BeneficiaryAppointmentItem__ConsultationInfo>
  </BeneficiaryAppointmentItem__ConsultationInfoContainer>
</BeneficiaryAppointmentItem__ConsultationRow>
`;
